agent/peering_endpoint.go

```diff
@@ -48,7 +48,7 @@ func (s *HTTPHandlers) peeringRead(resp http.ResponseWriter, req *http.Request,
 		return nil, err
 	}
 	if result.Peering == nil {
-		return nil, HTTPError{StatusCode: http.StatusNotFound}
+		return nil, HTTPError{StatusCode: http.StatusNotFound, Reason: fmt.Sprintf("peering with name \"%s\" does not exist", name)}
```

Comment:
I would prefer if we used the same error convention that we do for partitions/namespaces:
```suggestion
		return nil, HTTPError{StatusCode: http.StatusNotFound, Reason: fmt.Sprintf("Peering not found for %q", name)}
```